modules/nw-ptp-cloud-event-proxy-deployment-reference.adoc

```diff
@@ -0,0 +1,212 @@
+// Module included in the following assemblies:
+//
+// * networking/using-ptp.adoc
+
+:_content-type: REFERENCE
+[id="nw-ptp-cloud-event-proxy-deployment-reference_{context}"]
+= Cloud events consumer application developer reference
+
+When developing applications that make use of PTP or hardware events on a bare-metal cluster node, you need to deploy a `cloud-event-proxy` container alongside your consumer application in the application pod.
+The `cloud-event-proxy` container receives the events from the PTP Operator or {redfish-operator} pod and passes it to the consumer application.
+The `cloud-event-proxy` container uses REST API to subscribe to the events.
+
+[NOTE]
+====
+The following information provides general guidance for developing applications that use PTP or hardware events.
+You must have configured PTP or hardware events in the cluster.
+You must also configure `Namespace`, `ServiceAccount`, and `Role` custom resources (CRs) as required.
+====
+
+[discrete]
+== PTP events available from the cloud-event-proxy sidecar REST API
+
+.PTP events available from the cloud-event-proxy sidecar
+[options="header"]
+|====
+|Resource URI|Description
+|`/cluster/node/<node_name>/sync/ptp-status/lock-state`| Describes the current status of the PTP equipment lock state. Can be in `LOCKED`, `HOLDOVER` or `FREERUN` state.
+|`/cluster/node/<node_name>/sync/sync-status/os-clock-sync-state`| Describe the host operating system clock synchronization state. Can be in `LOCKED` or `FREERUN` state.
+|`/cluster/node/<node_name>/sync/ptp-status/ptp-clock-class-change`| Describes the current state of the PTP clock class.
+|====
+
+[discrete]
+== Consumer application web service
+
+Consumer applications require a web service running with a `POST` handler to receive the cloud native events JSON payload.
+The following example `server()` function in Go illustrates one potential approach.
+
+.Example events consumer server function in Go
+[source,go]
+----
+func server() {
+  http.HandleFunc("/event", getEvent)
+  http.HandleFunc("/ack/event", ackEvent)
+  http.ListenAndServe("localhost:8989", nil)
+}
+
+func getEvent(w http.ResponseWriter, req *http.Request) {
+  defer req.Body.Close()
+  bodyBytes, err := io.ReadAll(req.Body)
+  if err != nil {
+    log.Errorf("error reading event %v", err)
+  }
+  e := string(bodyBytes)
+  if e != "" {
+    processEvent(bodyBytes)
+    log.Infof("received event %s", string(bodyBytes))
+  } else {
+    w.WriteHeader(http.StatusNoContent)
+  }
+}
+
+func ackEvent(w http.ResponseWriter, req *http.Request) {
+  defer req.Body.Close()
+  bodyBytes, err := io.ReadAll(req.Body)
+  if err != nil {
+    log.Errorf("error reading acknowledgment  %v", err)
+  }
+  e := string(bodyBytes)
+  if e != "" {
+    log.Infof("received ack %s", string(bodyBytes))
+  } else {
+    w.WriteHeader(http.StatusNoContent)
+  }
+}
+----
```

Comment:
add service for exposing consumer side car service for HTTP transport( See shared  google doc)